In [2]:
import pandas as pd
df = pd.read_csv('NovaMarket_datos_crudos.csv')
df.head()

,id_pedido,id_cliente,fecha_compra,canal,ciudad,codigo_postal,categoria_producto,producto,precio,unidades,edad_cliente,correo,nivel_satisfaccion,fecha_actualizacion_stock
0,P00383,C0276,2026-01-03,App,Bucaramanga,68001,Belleza,Camiseta,-1742611.0,4,-19,c0276gmail.com,alto,2026-07-01
1,P00403,C0167,6/12/2026,Web,Cali,76001,Deportes,Cafetera,1828583.0,2,23,c0167@hotmail.com,MEDIO,2026-07-01
2,P00403,C0167,6/12/2026,Web,Cali,76001,Deportes,Cafetera,1828583.0,2,23,c0167@hotmail.com,MEDIO,2026-07-01
3,P00450,C0128,07/09/2026,App,bucaramanga,68001,electronica,Set bloques,787178.0,6,69,c0128@hotmail.com,MEDIO,2026-07-15
4,P00449,C0304,2025-07-19,Tienda,Bogotá,8001,Belleza,Audífonos BT,357231.0,5,67,c0304@novamarket.co,Bajo,2026-05-20


In [7]:
df.shape

(620, 14)

**¿Cuántos registros y variables hay?**
El dataset cuenta con 620 y 14 variables

In [10]:
df.head()

,id_pedido,id_cliente,fecha_compra,canal,ciudad,codigo_postal,categoria_producto,producto,precio,unidades,edad_cliente,correo,nivel_satisfaccion,fecha_actualizacion_stock
0,P00383,C0276,2026-01-03,App,Bucaramanga,68001,Belleza,Camiseta,-1742611.0,4,-19,c0276gmail.com,alto,2026-07-01
1,P00403,C0167,6/12/2026,Web,Cali,76001,Deportes,Cafetera,1828583.0,2,23,c0167@hotmail.com,MEDIO,2026-07-01
2,P00403,C0167,6/12/2026,Web,Cali,76001,Deportes,Cafetera,1828583.0,2,23,c0167@hotmail.com,MEDIO,2026-07-01
3,P00450,C0128,07/09/2026,App,bucaramanga,68001,electronica,Set bloques,787178.0,6,69,c0128@hotmail.com,MEDIO,2026-07-15
4,P00449,C0304,2025-07-19,Tienda,Bogotá,8001,Belleza,Audífonos BT,357231.0,5,67,c0304@novamarket.co,Bajo,2026-05-20


| Variable | Tipo | Subtipo |
|---|---|---|
| `id_pedido` | Cualitativa | Nominal (identificador, no cuantifica nada aunque sea numérico) |
| `id_cliente` | Cualitativa | Nominal (identificador) |
| `fecha_compra` | Cuantitativa* | Fecha/temporal (se puede tratar como discreta o como variable de tiempo) |
| `canal` | Cualitativa | Nominal (ej: web, tienda física, app — sin orden natural) |
| `ciudad` | Cualitativa | Nominal |
| `codigo_postal` | Cualitativa | Nominal (aunque son números, no tiene sentido sumarlos o promediarlos) |
| `categoria_producto` | Cualitativa | Nominal |
| `producto` | Cualitativa | Nominal |
| `precio` | Cuantitativa | Continua |
| `unidades` | Cuantitativa | Discreta (valores enteros contables) |
| `edad_cliente` | Cuantitativa | Discreta (en años completos, aunque conceptualmente es continua) |
| `correo` | Cualitativa | Nominal (identificador, texto libre) |
| `nivel_satisfaccion` | Cualitativa | **Ordinal** (si es tipo "bajo/medio/alto" o escala 1-5 con orden) |
| `fecha_actualizacion_stock` | Cuantitativa* | Fecha/temporal |


In [9]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 620 entries, 0 to 619
Data columns (total 14 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   id_pedido                  620 non-null    str    
 1   id_cliente                 620 non-null    str    
 2   fecha_compra               620 non-null    str    
 3   canal                      620 non-null    str    
 4   ciudad                     589 non-null    str    
 5   codigo_postal              620 non-null    int64  
 6   categoria_producto         620 non-null    str    
 7   producto                   620 non-null    str    
 8   precio                     599 non-null    float64
 9   unidades                   620 non-null    int64  
 10  edad_cliente               620 non-null    int64  
 11  correo                     516 non-null    str    
 12  nivel_satisfaccion         574 non-null    str    
 13  fecha_actualizacion_stock  620 non-null    str    
dtypes: fl

In [6]:
df.describe(include='all')

,id_pedido,id_cliente,fecha_compra,canal,ciudad,codigo_postal,categoria_producto,producto,precio,unidades,edad_cliente,correo,nivel_satisfaccion,fecha_actualizacion_stock
count,620,620,620,620,589,620.000000,620,620,5.990000e+02,620.000000,620.000000,516,574,620
unique,600,311,495,3,19,NaN,17,14,NaN,NaN,NaN,429,8,8
top,P00403,C0043,15/11/2027,Tienda,bucaramanga,NaN,Belleza,Perfume,NaN,NaN,NaN,c0012@hotmail.com,Alto,2026-07-15
freq,2,6,9,224,58,NaN,66,52,NaN,NaN,NaN,4,91,152
mean,NaN,NaN,NaN,NaN,NaN,31917.129032,NaN,NaN,4.888703e+06,188.682258,48.012903,NaN,NaN,NaN
std,NaN,NaN,NaN,NaN,NaN,30339.586863,NaN,NaN,4.292512e+07,1240.652475,28.668511,NaN,NaN,NaN
min,NaN,NaN,NaN,NaN,NaN,5001.000000,NaN,NaN,-2.448455e+06,-3.000000,-20.000000,NaN,NaN,NaN
25%,NaN,NaN,NaN,NaN,NaN,8001.000000,NaN,NaN,5.914020e+05,2.000000,31.000000,NaN,NaN,NaN
50%,NaN,NaN,NaN,NaN,NaN,13001.000000,NaN,NaN,1.178666e+06,3.000000,46.000000,NaN,NaN,NaN
75%,NaN,NaN,NaN,NaN,NaN,68001.000000,NaN,NaN,1.856670e+06,5.000000,63.000000,NaN,NaN,NaN


## 1. Valores faltantes (nulos)

| Columna | Nulos | % aprox |
|---|---|---|
| `correo` | 620 - 516 = **104** | ~16.8% |
| `nivel_satisfaccion` | 620 - 574 = **46** | ~7.4% |
| `precio` | 620 - 599 = **21** | ~3.4% |
| `ciudad` | 620 - 589 = **31** | ~5% |

## 2. Formatos raros / sospechosos por el dtype

- **`fecha_compra` y `fecha_actualizacion_stock` son `str`, no `datetime`**. -> Hay formatos mixtos 
- **`codigo_postal` es `int64`**. Riesgo típico: si hay códigos postales que empiezan en 0, al convertirlos a entero se pierde el cero inicial
- **`nivel_satisfaccion` es `str`**, lo cual sugiere que no es númerico (1-5) sino texto 